src/binary_format.h

```diff
@@ -33,31 +34,64 @@
 namespace dd {
 
 /**
+ * a few specialized functions for writing big endian values
+ */
+template <int num_bytes>
+inline void write_be(std::ostream &output, void *value);
+template <>
+inline void write_be<1>(std::ostream &output, void *value) {
+  output.write((char *)value, 1);
+}
+template <>
+inline void write_be<2>(std::ostream &output, void *value) {
+  uint16_t tmp = htobe16(*(uint16_t *)value);
+  output.write((char *)&tmp, sizeof(tmp));
+}
+template <>
+inline void write_be<4>(std::ostream &output, void *value) {
+  uint32_t tmp = htobe32(*(uint32_t *)value);
+  output.write((char *)&tmp, sizeof(tmp));
+}
+template <>
+inline void write_be<8>(std::ostream &output, void *value) {
+  uint64_t tmp = htobe64(*(uint64_t *)value);
+  output.write((char *)&tmp, sizeof(tmp));
+}
+
+/**
+ * a handy way to serialize values of certain type in big endian
+ */
+template <typename T>
+inline void write_be(std::ostream &output, T value) {
+  write_be<sizeof(T)>(output, &value);
```

Comment:
I think I'm going to hold off on the OOC stuff until this is done since I might need `write_be` and `read_be` as well.